sql/kvstore/kvstore.go

```diff
@@ -0,0 +1,62 @@
+package kvstore
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-scale"
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+func addKeyValue(db sql.Executor, key string, value scale.Encodable) error {
```

Comment:
I don't get why pass a `string` to later convert to `[]byte`. Why not just:
```suggestion
func addKeyValue(db sql.Executor, key []byte, value scale.Encodable) error {
```

Reply:
The reason is that `[]byte` cannot be `const` so we either have to do the conversation between `string` and `[]byte` here in this function or the caller of it has to convert.

I prefer to do the conversation here, to only have to do it once instead of everywhere this function is called.

Reply:
By using a `string`, the API has limited usability. The `string` type is a subset of `[]byte` type, isn't it? `string` can be converted to `[]byte`, but not the other way around.

Therefore, it will not be possible to use this API to store objects with raw byte keys (for example a hash).